dotCMS/src/main/java/com/dotcms/contenttype/business/BaseTypeToContentTypeStrategyResolver.java

```diff
@@ -0,0 +1,120 @@
+ */
+public class BaseTypeToContentTypeStrategyResolver {
+
+    private volatile Map<BaseContentType, BaseTypeToContentTypeStrategy> strategiesMap = this.getDefaultStrategies();
+
+    private  Map<BaseContentType, BaseTypeToContentTypeStrategy> getDefaultStrategies() {
+
+        final ImmutableMap.Builder<BaseContentType, BaseTypeToContentTypeStrategy> builder =
+                new ImmutableMap.Builder<>();
+
+        builder.put(BaseContentType.DOTASSET, new DotAssetBaseTypeToContentTypeStrategyImpl());
+
+        return builder.build();
+    }
+
+
+    public synchronized void subscribe (final BaseContentType baseContentType, final BaseTypeToContentTypeStrategy strategy) {
+
+        if (null != baseContentType && null != strategy) {
+
+            final ImmutableMap.Builder<BaseContentType, BaseTypeToContentTypeStrategy> builder =
+                    new ImmutableMap.Builder<>();
+
+            builder.putAll(this.strategiesMap);
+            builder.put(BaseContentType.DOTASSET, strategy);
+
+            this.strategiesMap = builder.build();
+        }
+    }
+
+
+    private static class SingletonHolder {
+        private static final BaseTypeToContentTypeStrategyResolver INSTANCE = new BaseTypeToContentTypeStrategyResolver();
+    }
+
+    /**
+     * Get the instance.
+     * @return BaseTypeToContentTypeStrategyResolver
+     */
+    public static BaseTypeToContentTypeStrategyResolver getInstance() {
+
+        return BaseTypeToContentTypeStrategyResolver.SingletonHolder.INSTANCE;
+    } // getInstance.
+
+
+    /**
+     * Get a strategy if applies
+     * @param baseContentType {@link BaseContentType}
+     * @return Optional BaseTypeToContentTypeStrategy
+     */
+    public Optional<BaseTypeToContentTypeStrategy> get(final BaseContentType baseContentType) {
+
+        return this.strategiesMap.containsKey(baseContentType)?
+                Optional.of(this.strategiesMap.get(baseContentType)):
+                Optional.empty();
+    }
+    /////////////
+    private class DotAssetBaseTypeToContentTypeStrategyImpl  implements BaseTypeToContentTypeStrategy {
+
+        @Override
+        public Optional<ContentType> apply(final BaseContentType baseContentType, final Map<String, Object> contextMap) {
+
+            final User user = (User)contextMap.get("user");
+            final Host currentHost = (Host)contextMap.get("host");
+            final List<File> binaryFiles = (List<File>) contextMap.getOrDefault("binaryFiles", Collections.emptyList());
+            final Map<String, Object> contentletMap = (Map<String, Object>) contextMap.get("contentletMap");
+
+            final File file = this.getBinary(binaryFiles, contentletMap);
+            if (null != file && file.exists() && file.canRead()) {
+
+                try {
+
+                    return APILocator.getDotAssetAPI().tryMatch(file, currentHost, user);
+                } catch (DotDataException | DotSecurityException e) {
+                    return Optional.empty();
```

Comment:
I think this return here is redundant as there is a `return Optional.empty()` at the end of the method. Besides, it might be worth to a log at least of debug level